homework/index.js

```diff
@@ -30,14 +30,120 @@
     return elem;
   }
 
+  function clearContainer(container) {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function renderError(err) {
+    const listContainer = document.getElementById('list-container');
+    clearContainer(listContainer);
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert alert-error' });
+  }
+
+  function addRow(tbody, label, repository) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:`, class: 'label' });
+    if (label === 'Repository:') {
+      const urlRepo = repository.html_url;
+      const td = createAndAppend('td', tr);
+      createAndAppend('a', td, {
+        text: repository.name,
+        href: urlRepo,
+        target: '_blank',
+      });
+    }
+    if (label === 'Description:') {
+      createAndAppend('td', tr, { text: repository.description });
+      console.log(repository.description);
+    }
+    if (label === 'Forks:') {
+      createAndAppend('td', tr, { text: repository.forks });
+    }
+
+    if (label === 'Updated:') {
+      createAndAppend('td', tr, { text: repository.updated_at });
+    }
+  }
+
+  function createInfoList(repository, container) {
+    clearContainer(container);
+    const div = createAndAppend('div', container, { id: 'info_div' });
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    createAndAppend('tr', table);
+    createAndAppend('tr', table, {});
+    addRow(tbody, 'Repository:', repository);
+    addRow(tbody, 'Description:', repository);
+    addRow(tbody, 'Forks:', repository);
+    addRow(tbody, 'Updated:', repository);
+  }
+
+  function createContributorsList(repositoryName, ul) {
+    clearContainer(ul);
+    const url = `https://api.github.com/repos/HackYourFuture/${repositoryName}/contributors`;
```

Comment:
Do not construct this url yourself. Instead use the `.contributors_url` property of the repository object, as per the hint from the assignment. The reason for this is that GitHub may choose to change the internal structure of the url in the future. However, it will continue to support the documented `.contributors_url` property.